components/net_traits/bluetooth_scanfilter.rs

```diff
@@ -69,13 +72,13 @@ impl BluetoothScanfilter {
     }
 
     pub fn is_empty_or_invalid(&self) -> bool {
-        (self.name.is_empty() &&
+        (self.name.is_none() &&
          self.name_prefix.is_empty() &&
          self.get_services().is_empty() &&
          self.manufacturer_id.is_none() &&
          self.service_data_uuid.is_empty()) ||
-        self.name.len() > MAX_NAME_LENGTH ||
-        self.name_prefix.len() > MAX_NAME_LENGTH
+         self.get_name().unwrap_or("").len() > MAX_NAME_LENGTH ||
```

Comment:
These lines should be unindented by one, since they are not part of the subexpression.